### ISSUE TITLE
POST request with path and body params gives error

### ISSUE THREAD

**Issue by sadath-12** [nineinchnick/trino-openapi#111]
When I run the query
```sh
trino> SELECT * 
    -> FROM openapi.default.namespaces_indexes_search 
    -> WHERE namespace = 'default' 
    ->   AND index = 'rag_pipeline6.wikiembedding.embedding' 
    ->   AND query = 'whats in content'  AND k=1 AND include_content=true;
```

```
Query 20240804_141803_00021_xa39m, FAILED, 1 node
http://localhost:8080/ui/query.html?20240804_141803_00021_xa39m
Splits: 1 total, 0 done (0.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 0% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 0B
0.02 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20240804_141803_00021_xa39m failed: Cannot invoke "pl.net.was.OpenApiColumn.getSourceName()" because "column" is null
java.lang.NullPointerException: Cannot invoke "pl.net.was.OpenApiColumn.getSourceName()" because "column" is null
        at pl.net.was.OpenApiClient.serializeMap(OpenApiClient.java:377)
        at pl.net.was.OpenApiClient.getRows(OpenApiClient.java:137)
        at pl.net.was.OpenApiRecordSetProvider.getRecordSet(OpenApiRecordSetProvider.java:74)
        at io.trino.split.RecordPageSourceProvider.createPageSource(RecordPageSourceProvider.java:50)
        at io.trino.split.PageSourceManager$PageSourceProviderInstance.createPageSource(PageSourceManager.java:79)
        at io.trino.operator.TableScanOperator.getOutput(TableScanOperator.java:265)
        at io.trino.operator.Driver.processInternal(Driver.java:403)
        at io.trino.operator.Driver.lambda$process$8(Driver.java:306)
        at io.trino.operator.Driver.tryWithLock(Driver.java:709)
        at io.trino.operator.Driver.process(Driver.java:298)
        at io.trino.operator.Driver.processForDuration(Driver.java:269)
        at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:890)
        at io.trino.execution.executor.dedicated.SplitProcessor.run(SplitProcessor.java:77)
        at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.lambda$run$0(TaskEntry.java:191)
        at io.trino.$gen.Trino_450____20240804_140436_2.run(Unknown Source)
        at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.run(TaskEntry.java:192)
        at io.trino.execution.executor.scheduler.FairScheduler.runTask(FairScheduler.java:168)
        at io.trino.execution.executor.scheduler.FairScheduler.lambda$submit$0(FairScheduler.java:155)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1570)


```
The table looks like : 
<img width="1224" alt="Screenshot 2024-08-04 at 7 50 10 PM" src="https://github.com/user-attachments/assets/632435dc-98d4-4448-b866-8d42fde3f277">

Equivalent curl request 

```sh
curl -X 'POST' \
  'http://ip:port/namespaces/default/indexes/rag_pipeline6.wikiembedding.embedding/search' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{

  "include_content": true,
  "k": 1,
  "query": "whats in content"
}'
```

**Comment by sadath-12** [comment]
sample expected response from this 

```yaml
xample Value
Schema
{
  "results": [
    {
      "confidence_score": 0,
      "content_id": "string",
      "content_metadata": {
        "created_at": 0,
        "extracted_metadata": "string",
        "extraction_graph_names": [
          "string"
        ],
        "hash": "string",
        "id": "string",
        "labels": {
          "additionalProp1": "string",
          "additionalProp2": "string",
          "additionalProp3": "string"
        },
        "mime_type": "string",
        "name": "string",
        "namespace": "string",
        "parent_id": "string",
        "root_content_id": "string",
        "size": 0,
        "source": "string",
        "storage_url": "string"
      },
      "labels": {
        "additionalProp1": "string",
        "additionalProp2": "string",
        "additionalProp3": "string"
      },
      "mime_type": "string",
      "root_content_metadata": {
        "created_at": 0,
        "extracted_metadata": "string",
        "extraction_graph_names": [
          "string"
        ],
        "hash": "string",
        "id": "string",
        "labels": {
          "additionalProp1": "string",
          "additionalProp2": "string",
          "additionalProp3": "string"
        },
        "mime_type": "string",
        "name": "string",
        "namespace": "string",
        "parent_id": "string",
        "root_content_id": "string",
        "size": 0,
        "source": "string",
        "storage_url": "string"
      },
      "text": "string"
    }
  ]
}
```

**Comment by nineinchnick** [comment]
This is caused by not searching all columns when copying predicates back to query results: https://github.com/nineinchnick/trino-openapi/blob/main/src/main/java/pl/net/was/OpenApiClient.java#L362

Since 1134a7f0fde09327418baeb2adbfa46ef06d9f1a, path params in POST requests are correctly mapped to columns, but only body params are included in the method I referenced above.

**Comment by sadath-12** [comment]
@nineinchnick  so how can I make this work now ?

**Comment by sadath-12** [comment]
would be great , if I can get the query example

**Comment by nineinchnick** [comment]
Should be fixed in bf76673e96adede9cbef2ee0bb615e57ad601f51. If it's not, please reopen the issue.

**Comment by sadath-12** [comment]
works now , thanks
